tests/merch.order.test.ts

```diff
@@ -0,0 +1,721 @@
+      .thenResolve();
+
+    // place order
+    const order = [
+      {
+        option: affordableOption.uuid,
+        quantity: 1,
+      },
+    ];
+    const placeMerchOrderRequest = {
+      order,
+      pickupEvent: pickupEvent.uuid,
+    };
+    const merchController = ControllerFactory.merchStore(conn, instance(emailService));
+    const placedOrderResponse = await merchController.placeMerchOrder(placeMerchOrderRequest, member);
+
+    // cancel order
+    const { uuid } = placedOrderResponse.order;
+    expect(merchController.cancelMerchOrder({ uuid }, otherMember))
+      .rejects
+      .toThrow('Member cannot cancel other members\' orders');
+  });
+
+  test('admins can fulfill parts of a member\'s order', async () => {
+    const conn = await DatabaseConnection.get();
+    const member = UserFactory.fake({ credits: 10000 });
+    const admin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+    const affordableOption1 = MerchFactory.fakeOption({
+      quantity: 1,
+      price: 2000,
+      discountPercentage: 0,
+    });
+    const affordableOption2 = MerchFactory.fakeOption({
+      quantity: 1,
+      price: 3000,
+      discountPercentage: 0,
+    });
+    const pickupEvent = MerchFactory.fakeFutureOrderPickupEvent();
+
+    await new PortalState()
+      .createUsers(member)
+      .createMerchItemOption(affordableOption1)
+      .createMerchItemOption(affordableOption2)
+      .createOrderPickupEvents(pickupEvent)
+      .write();
+
+    const emailService = mock(EmailService);
+    when(emailService.sendOrderConfirmation(member.email, member.firstName, anything()))
+      .thenResolve();
+
+    // place order
+    const order = [
+      {
+        option: affordableOption1.uuid,
+        quantity: 1,
+      },
+      {
+        option: affordableOption2.uuid,
+        quantity: 1,
+      },
+    ];
+    const placeMerchOrderRequest = {
+      order,
+      pickupEvent: pickupEvent.uuid,
+    };
+    const merchController = ControllerFactory.merchStore(conn, instance(emailService));
+    const placedOrderResponse = await merchController.placeMerchOrder(placeMerchOrderRequest, member);
+
+    // fulfill a single item in an order
+    const placedOrder = placedOrderResponse.order;
+    const uuidParams = { uuid: placedOrder.uuid };
+    const fulfillMerchOrderItemsRequest = {
+      items: [
+        {
+          uuid: placedOrder.items[0].uuid,
+          fulfilled: true,
+        },
+        {
+          uuid: placedOrder.items[1].uuid,
+          fulfilled: false,
+        },
+      ],
+    };
+    await merchController.fulfillMerchOrderItems(uuidParams, fulfillMerchOrderItemsRequest, admin);
+
+    const getOrderResponse = await merchController.getOneMerchOrder(uuidParams, member);
+    const partiallyFulfilledOrder = getOrderResponse.order;
+
+    // order status should not be FULFILLED because entire order isn't fulfilled
+    expect(partiallyFulfilledOrder.status).toEqual(OrderStatus.PLACED);
```

Comment:
PLACED doesn't seem like the correct state for this. Maybe PARTIALLY_FULFILLED? Like, what if an order is partially fulfilled, and then the user tries to get a full refund, and we say "ok your order's PLACED but not FULFILLED so sure here's your full refund even though you already got some item(s)." Let's add a test for getting refunds on a partially fulfilled order.

Reply:
I don't think we'd actually need `PARTIALLY_FULFILLED` since we can just refund all the items of an order where `fulfilled = false`, right? Since `PARTIALLY_FULFILLED` is just a state for if any item has been fulfilled already.

Reply:
Yeah, but it's not exactly accurate and might be a foot gun if someone down the road thinks `PLACED` means `NONE_FULFILLED`. I think it's an important distinction between none, any, and all fulfilled.